argon2/src/block.rs

```diff
@@ -67,6 +71,55 @@ impl Block {
     }
 
     pub(crate) fn compress(rhs: &Self, lhs: &Self) -> Self {
+        #[cfg(any(target_arch = "x86_64"))]
+        {
+            let (_, avx2) = avx2_cpuid::init_get();
+            if avx2 {
+                return unsafe { Self::compress_avx2(rhs, lhs) };
+            }
+        }
+        Self::compress_safe(rhs, lhs)
+    }
+
+    fn compress_safe(rhs: &Self, lhs: &Self) -> Self {
```

Comment:
We more typically use `soft` or `portable` to refer to these kinds of intrinsic-free implementations, rather than `safe`